mne/preprocessing/ica.py

```diff
@@ -544,16 +544,14 @@ def _fit(self, data, max_pca_components, fit_type):
         """Aux function."""
         random_state = check_random_state(self.random_state)
 
+        from sklearn.decomposition import PCA
         if not check_version('sklearn', '0.18'):
-            from sklearn.decomposition import RandomizedPCA
-            # XXX fix copy==True later. Bug in sklearn, see PR #2273
-            pca = RandomizedPCA(n_components=max_pca_components, whiten=True,
-                                copy=True, random_state=random_state)
-
+            # fix copy==True later. Bug in sklearn, see PR #2273
+            pca = PCA(n_components=max_pca_components, whiten=True, copy=True,
+                      random_state=random_state)
         else:
-            from sklearn.decomposition import PCA
-            pca = PCA(n_components=max_pca_components, copy=True, whiten=True,
-                      svd_solver='randomized', random_state=random_state)
+            pca = PCA(n_components=max_pca_components, whiten=True, copy=True,
+                      svd_solver='full', random_state=random_state)
```

Comment:
yes they are legit, `PCA` has no `random_state` argument before 0.18

http://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html

But it should only be needed if using the `randomized` solver anyway. So `mne.utils.get_args` can be used to inspect if there is a `svd_solver` arg and if so, set to `'full'` (and ignore ICA) otherwise don't need to set it

Reply:
... oh we already have a version check. So just change the call appropriately